server/src/main/java/com/broadcom/lsp/cobol/core/model/VarFinder.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2020 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+package com.broadcom.lsp.cobol.core.model;
+
+import com.broadcom.lsp.cobol.core.model.variables.Variable;
+import com.google.common.collect.Lists;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.groupingBy;
+
+/**
+ * The class take all defined variables and search thought them by partial qualifier
+ */
+public class VarFinder {
+  private final Map<String, List<Variable>> definedVariables;
+
+  public VarFinder(Collection<Variable> definedVariables) {
+    this.definedVariables = definedVariables.stream().collect(groupingBy(Variable::getName));
```

Comment:
My bad, I didn't see where this grouping is actually called from.
Ok, that makes sense not to group it in the definition delegate, but still, I think it would be better to avoid storing a state for that class. You may keep that map local in the `VariableUsageDelegate.updateUsageAndGenerateErrors` method